### PR TITLE
Move modals to querystring

### PR DESCRIPTION
**Pull request by dominik-zeglen**
I want to merge this change because it prepares the ground for the #3537 

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [ ] Database migration files are up to date.
1. [ ] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [ ] GraphQL schema and type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### CODE REVIEW

File: saleor/static/dashboard-next/collections/views/CollectionDetails.tsx [hunk 1]
```diff
@@ -24,362 +22,310 @@ import { CollectionUpdate } from "../types/CollectionUpdate";
...
-                            title: formData.seoTitle
-                          }
-                        };
-                        const isFeatured = data.shop.homepageCollection
-                          ? data.shop.homepageCollection.id ===
-                            data.collection.id
-                          : false;
+                const isFeatured = data.shop.homepageCollection
+                  ? data.shop.homepageCollection.id === data.collection.id
+                  : false;
```
Comment (piotrgrundas) [inline saleor/static/dashboard-next/collections/views/CollectionDetails.tsx]: Smells like `maybe`

File: saleor/static/dashboard-next/customers/views/CustomerAddresses.tsx [hunk 2]
```diff
@@ -140,140 +141,99 @@ const CustomerAddresses: React.FC<CustomerAddressesProps> = ({ id }) => {
...
-                                          addr.id ===
-                                          decodeURIComponent(
-                                            match.params.addressId
-                                          )
-                                      )
-                                    )}
-                                    confirmButtonState={
-                                      updateAddressTransitionState
+                            <CustomerAddressDialog
+                              address={undefined}
```
Comment (piotrgrundas) [inline saleor/static/dashboard-next/customers/views/CustomerAddresses.tsx]: Does it need be explicit? Mb can be set as not required on `CustomerAddressDialog` if so?

File: saleor/static/dashboard-next/customers/views/CustomerAddresses.tsx [hunk 3]
```diff
@@ -140,140 +141,99 @@ const CustomerAddresses: React.FC<CustomerAddressesProps> = ({ id }) => {
...
                                     }
-                                  />
+                                  }
+                                })
+                              }
+                            />
+                            <ActionDialog
+                              open={params.action === "remove"}
+                              variant="delete"
+                              title={i18n.t("Remove Address")}
```
Comment (piotrgrundas) [inline saleor/static/dashboard-next/customers/views/CustomerAddresses.tsx]: Just wondering - wont be easier to translate `title` inside `ActionDialog`? 
I'm assuming there won't be situations when u don't intend to translate it.

File: saleor/static/dashboard-next/customers/views/CustomerDetails.tsx [hunk 4]
```diff
@@ -20,154 +19,142 @@ import { UpdateCustomer } from "../types/UpdateCustomer";
...
+                    />
+                    <ActionDialog
+                      confirmButtonState={removeTransitionState}
+                      onClose={() => navigate(customerUrl(id), true)}
+                      onConfirm={() => removeCustomer()}
+                      title={i18n.t("Remove customer", {
+                        context: "modal title"
+                      })}
+                      variant="delete"
+                      open={params.action === "remove"}
```
Comment (piotrgrundas) [inline saleor/static/dashboard-next/customers/views/CustomerDetails.tsx]: IMO should be an `Enum`.

File: saleor/static/dashboard-next/discounts/views/SaleDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,392 @@
...
+                                })
+                              }
+                              onRemove={() => openModal("remove")}
+                              saveButtonBarState={formTransitionState}
+                            />
+                            <SearchProductsProvider>
+                              {(searchProducts, searchProductsOpts) => (
+                                <AssignProductDialog
+                                  confirmButtonState={assignTransitionState}
+                                  open={params.action === "assign-product"}
```
Comment (piotrgrundas) [inline saleor/static/dashboard-next/discounts/views/SaleDetails.tsx]: `Enum`

File: saleor/static/dashboard-next/orders/views/OrderDetails/index.tsx [hunk 6]
```diff
@@ -546,253 +451,213 @@ export const OrderDetails: React.StatelessComponent<OrderDetailsProps> = ({
...
+                                orderUpdate.mutate({
+                                  id,
+                                  input: {
+                                    billingAddress: {
+                                      ...variables,
+                                      country: variables.country.value
+                                    }
+                                  }
+                                })
+                              }
```
Comment (piotrgrundas) [inline saleor/static/dashboard-next/orders/views/OrderDetails/index.tsx]: This should be splited.

File: saleor/static/dashboard-next/products/views/ProductImage.tsx [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+                    navigate(productImageUrl(productId, id));
+                  const handleUpdate = (formData: { description: string }) => {
+                    updateImage({
+                      variables: {
+                        alt: formData.description,
+                        id: imageId
+                      }
+                    });
+                  };
+                  const image = data && data.product && data.product.mainImage;
```
Comment (piotrgrundas) [inline saleor/static/dashboard-next/products/views/ProductImage.tsx]: use `maybe`

File: saleor/static/dashboard-next/shipping/views/ShippingZoneDetails/ShippingZoneDetailsDialogs.tsx [hunk 8]
```diff
@@ -45,282 +36,205 @@ const ShippingZoneDetailsDialogs: React.StatelessComponent<
...
+                  ? formData.noLimits
+                    ? null
+                    : parseFloat(formData.minValue)
+                  : null,
+              minimumOrderWeight:
+                params.type === ShippingMethodTypeEnum.WEIGHT
+                  ? formData.noLimits
+                    ? null
+                    : parseFloat(formData.minValue)
+                  : null,
```
Comment (piotrgrundas) [inline saleor/static/dashboard-next/shipping/views/ShippingZoneDetails/ShippingZoneDetailsDialogs.tsx]: xxx = (type, value) => 
    params.type === type
        ? formData.noLimits
          ? null
          : parseFloat(value)
        : null

File: saleor/static/dashboard-next/siteSettings/views/index.tsx [hunk 9]
```diff
@@ -23,191 +22,180 @@ import { TypedSiteSettingsQuery } from "../queries";
...
+          context: "notification",
+          message: data.authorizationKeyDelete.errors[0].message
+        })
+      });
+    }
+  };
+  const handleSiteSettingsSuccess = (data: ShopSettingsUpdate) => {
+    if (
+      (!data.shopDomainUpdate.errors ||
+        data.shopDomainUpdate.errors.length === 0) &&
```
Comment (piotrgrundas) [inline saleor/static/dashboard-next/siteSettings/views/index.tsx]: If API does not always return a list the should fix it, it's not a big change.

File: saleor/static/dashboard-next/customers/views/CustomerAddresses.tsx (hunk 3 above)
Comment (dominik-zeglen) [reply saleor/static/dashboard-next/customers/views/CustomerAddresses.tsx]: Literals, man. Unfortunately.

File: saleor/static/dashboard-next/customers/views/CustomerDetails.tsx (hunk 4 above)
Comment (dominik-zeglen) [reply saleor/static/dashboard-next/customers/views/CustomerDetails.tsx]: Why?

File: saleor/static/dashboard-next/orders/views/OrderDetails/index.tsx (hunk 6 above)
Comment (dominik-zeglen) [reply saleor/static/dashboard-next/orders/views/OrderDetails/index.tsx]: That's not what this PR is about, and this view is waiting for the refactor anyway.

File: saleor/static/dashboard-next/siteSettings/views/index.tsx (hunk 9 above)
Comment (dominik-zeglen) [reply saleor/static/dashboard-next/siteSettings/views/index.tsx]: Old code, didn't even read this part.

**Pull request merged**